### ISSUE TITLE
🐛 noPrivateImports false positives for imports from node_modules in 2.0.0-beta.6

### ISSUE THREAD

**Issue by minht11** [biomejs/biome#6214]
### Environment information

```block
CLI:
  Version:                      2.0.0-beta.6
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  BIOME_THREADS:                unset
  NO_COLOR:                     unset
  TERM:                         xterm-256color
  JS_RUNTIME_VERSION:           v22.14.0
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         pnpm/10.10.0

Biome Configuration:
  Status:                       Loaded successfully
  Path:                         biome.jsonc
  Formatter enabled:            true
  Linter enabled:               true
  Assist enabled:               true
  VCS enabled:                  true

Workspace:
  Open Documents:
```

### What happened?

After upgrading from Biome 2.0.0-beta.5 to 2.0.0-beta.6 `noPrivateImports` rule started to get false positives for imports from node_modules. See path in photo which references pnpm node modules.
<img width="758" alt="Image" src="https://github.com/user-attachments/assets/bac6dffb-e4b5-4a07-bcbe-38c2b40c79b5" />

I assume expanded scanner support caused it and `.get_import_path_by_js_node(node)` now returns node_modules which before it didnt? https://github.com/biomejs/biome/blob/d42c38ca437f12a47651e98cf9108fff31860ac1/crates/biome_js_analyze/src/lint/correctness/no_private_imports.rs#L230-L232

### Expected result

No error reported for imports from node_modules

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by arendjr** [comment]
What is your default visibility setting for `noPrivateImports`?

**Comment by minht11** [comment]
I set visibility to "package".
Full [biome.jsonc](https://github.com/minht11/local-music-pwa/blob/next/biome.jsonc)

**Comment by minht11** [comment]
I installed latest prerelease version `https://pkg.pr.new/biomejs/biome/@biomejs/biome@0943777` and still experiencing false positives from packages, interestingly it no longer flags `import { getContext } from 'svelte'` but does `import { untrack } from 'svelte'`

[Reproduction](https://github.com/minht11/local-music-pwa/tree/experiment/biome-preview%400943777) with `biome@0943777`, you can run `pnpm run biome-check`

**Comment by arendjr** [comment]
@minht11 My bad! This should do the trick: https://github.com/biomejs/biome/pull/6327
